core/coreapi/unixfs.go

```diff
@@ -126,6 +131,23 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.File, opts ...options
 		fileAdder.SetMfsRoot(mr)
 	}
 
+	// Get storageMax when node is online.
+	if fileAdder.Pin && !n.EnableGC && n.OnlineMode() {
```

Comment:
Do we need to check for `Pin` here? the file will be added anyway even if we don't pin it, right? (it can be removed later but it will make the repo grow)